analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -811,7 +834,202 @@ public void M()
                     _ = {{memberAccessExpression}};
                 }
             }
-            """/*, allowSemanticModel */);
+            """, allowSemanticModel);
+
+    [DataTestMethod]
+    [DataRow("_ = [u:i];")]
+    [DataRow("[u:i] = [u:i] + [u:i];")]
+    [DataRow("_ = i.[u:ToString]().[u:ToString]();")] // "i" must be queried. It could be a type.
+    [DataRow("_ = [u:ex]?.[u:ToString]();")]
+    [DataRow("_ = ([u:ex] ?? new Exception()).[u:ToString]();")]
+    [DataRow("[u:ex] ??= new Exception();")]
+    [DataRow("_ = String.Format([k:string].[u:Empty], [u:i]);")]
+    [DataRow("_ = [u:ex] is ArgumentException { };")]
+    [DataRow("_ = [u:i] == [u:i] ? [u:b] : ![u:b];")]
+    [DataRow("if ([u:i] == [u:i]) { }")]
+    [DataRow("if ([u:b]) { }")]
+    [DataRow("switch ([u:i]) { }")]
+    [DataRow("_ = [u:i] switch { _ => true };")]
+    [DataRow("foreach (var [u:e] in [u:l]) { }")]
+    [DataRow("for(int [u:x] = 0; [u:b]; [u:x]++) { }")]
+    [DataRow("while([u:b]) { }")]
+    [DataRow("do i++; while([u:b]);")]
+    [DataRow("_ = l.[u:Where]([u:x] => [u:x] == null);")]
+    [DataRow("_ = ([u:i], [u:i]);")]
+    [DataRow("_ = int.TryParse(string.Empty, out [u:i]);")]
+    [DataRow("_ = new int[[u:i]];")]
+    [DataRow("await [u:t];")]
+    [DataRow("_ = unchecked([u:i] + [u:i]);")]
+    [DataRow("_ = [u:l][[u:i]];")]
+    [DataRow("_ = (byte)([u:i]);")]
+    [DataRow("_ = new { [u:A] = [u:i] };")]
+    [DataRow("""
+        _ = from [u:x] in [u:l]
+            select x;
+        """)]
+    [DataRow("""
+        _ = from [u:x] in [u:l]
+            let [u:y] = [u:x]
+            join [u:z] in [u:l] on [u:x] equals [u:z]
+            where [u:x] == [u:z]
+            orderby [u:x], [u:z]
+            select new { [u:x], Y = [u:y] };
+        """)]
+    [DataRow("_ = ([u:i]);")]
+    [DataRow("_ = +[u:i];")]
+    [DataRow("_ = ++[u:i];")]
+    [DataRow("_ = -[u:i];")]
+    [DataRow("_ = --[u:i];")]
+    [DataRow("_ = ~[u:i];")]
+    [DataRow("_ = ![u:b];")]
+    [DataRow("_ = [u:i]++;")]
+    [DataRow("_ = [u:i]--;")]
+    [DataRow("_ = [u:l]!;")]
+    [DataRow("_ = [u:i] + [u:i];")]
+    [DataRow("_ = [u:i] - [u:i];")]
+    [DataRow("_ = [u:i] / [u:i];")]
+    [DataRow("_ = [u:i] * [u:i];")]
+    [DataRow("_ = [u:i] % [u:i];")]
+    [DataRow("_ = [u:i] >> [u:i];")]
+    [DataRow("_ = [u:i] << [u:i];")]
+    [DataRow("_ = [u:b] && [u:b];")]
+    [DataRow("_ = [u:b] || [u:b];")]
+    [DataRow("_ = [u:i] & [u:i];")]
+    [DataRow("_ = [u:i] | [u:i];")]
+    [DataRow("_ = [u:i] ^ [u:i];")]
+    [DataRow("_ = [u:i] == [u:i];")]
+    [DataRow("_ = [u:i] != [u:i];")]
+    [DataRow("_ = [u:i] < [u:i];")]
+    [DataRow("_ = [u:i] <= [u:i];")]
+    [DataRow("_ = [u:i] > [u:i];")]
+    [DataRow("_ = [u:i] >= [u:i];")]
+    [DataRow("_ = [u:i] is iConst;")] // iConst could be a type
```

Comment:
Shouldn't be a TODO then? Now there are no queries for this.

Reply:
See the IdentifierToken_SimpleMemberAccess_Is test